packages/client/wallets/aa/src/services/logging/index.ts

```diff
@@ -1,16 +1,49 @@
 import { isLocalhost } from "@/utils/helpers";
 
-import { ConsoleProvider } from "./ConsoleProvider";
 import { DatadogProvider } from "./DatadogProvider";
 
-function getBrowserLogger() {
-    if (isLocalhost()) {
-        return new ConsoleProvider();
+class LoggerProvider {
+    private static instance: LoggerProvider;
+    public logInfo!: (message: string, context?: object) => void;
+    public logWarn!: (message: string, context?: object) => void;
+    public logError!: (message: string, context?: object) => void;
+
+    private constructor(private logOnDatadogOnLocalhost = false) {
+        this.setLogger();
     }
 
-    return new DatadogProvider();
-}
+    private setLogger() {
+        const logger = this.shouldLogToDatadog() ? new DatadogProvider() : this.getEmptyLogger();
+        const { logInfo, logWarn, logError } = logger;
+        this.logInfo = logInfo;
+        this.logWarn = logWarn;
+        this.logError = logError;
+    }
 
-const { logInfo, logWarn, logError } = getBrowserLogger();
+    private shouldLogToDatadog() {
+        return (isLocalhost() && this.logOnDatadogOnLocalhost) || !isLocalhost();
+    }
+
+    private getEmptyLogger() {
+        return {
+            logInfo: () => {},
+            logWarn: () => {},
+            logError: () => {},
+        };
+    }
+
+    public setDDLoggerOnLocalhost(logOnDatadog: boolean) {
+        console.log("setDDLoggerOnLocalhost", logOnDatadog);
+        this.logOnDatadogOnLocalhost = logOnDatadog;
+        this.setLogger();
+    }
+
+    public static getInstance() {
+        if (!LoggerProvider.instance) {
+            LoggerProvider.instance = new LoggerProvider();
+        }
+        return LoggerProvider.instance;
+    }
+}
 
-export { logInfo, logWarn, logError };
+export const logger = LoggerProvider.getInstance();
```

Comment:
I've got a bunch of thoughts here:
1. I don't think `export const logger` adds value, we already have the singleton set up, so I think better to just allow people to use `LoggerProvider.getLogger()`–this way the rest of the code looks more natural w/ dependency injection
2. `logOnDatadogOnLocalhost`–doesn't seem like the pattern we want relative to being able to provide a logger
3. Rather than this `LoggerProvider` having all the `logInfo`, `logWarning` set on itself, it'd be cleaner if it lived up to its name, and just provided a `Logger`, with the ability to set what that logger is. So together I'd be thinking more like:

```
interface Logger {
    logInfo: (message: string, context?: object) => void;
    logWarn: (message: string, context?: object) => void;
    logError: (message: string, context?: object) => void;
}

export class LoggerProvider {
    private static instance: LoggerProvider;
    private logger: Logger

    private constructor() {
        const defaultLogger = this.shouldLogToDatadog() ? new DatadogProvider() : new EmptyLogger()
        this.logger = defaultLogger;
    }

    public static setLogger(logger: Logger) {
        LoggerProvider.instance.logger = logger;
    }

    public static getLogger() {
        return LoggerProvider.instance.logger!
    }

    private shouldLogToDatadog() {
        return !isLocalhost();
    }

    public static getInstance() {
        if (!LoggerProvider.instance) {
            LoggerProvider.instance = new LoggerProvider();
        }
        return LoggerProvider.instance;
    }
}

class EmptyLogger implements Logger {
    logInfo()  {}
    logWarn()  {}
    logError()  {}
}